.github/workflows/makefile.yml

```diff
@@ -0,0 +1,37 @@
+name: Makefile CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: Set up Python 3.10.3
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10.3" # unfortunately we can't read from a file here, otherwise runtime.txt would be used
+
+      - name: Install Flit
+        run: python3 -m pip install flit
+
+      - name: Install dependencies
+        run: flit install
```

Comment:
Why not just,
```
pip install .
```

Reply:
Oh right. There's a bit of organization to the deps. "pip install .[test]" should install everything we need.

Reply:
`pip install .[test]` I don't think works without a `setup.py`, which flit is supposed to generate for us. When I try to run that locally I get - 

```
(venv) gseyffert@Grahams-MacBook-Pro spark_log_parser % pip install .[test]
zsh: no matches found: .[test]
```

Reply:
(we can of course _add_ a `setup.py`, but then it kind of begs the question of "then why use flit at all?". though that may just be a question that's coming from a place of ignorance on my end)

Reply:
That's a ZSH error. Try,
```
 % pip install '.[test]'
```
(you might not need the quotes in the action spec)

Reply:
ah, right you are - I completely misread that message! Updated :+1: